addons/test_mail/tests/test_mail_template.py

```diff
@@ -54,6 +54,65 @@ def test_template_send_email(self):
         self.assertEqual(mail.email_cc, self.email_template.email_cc)
         self.assertEqual(mail.recipient_ids, self.partner_2 | self.user_admin.partner_id)
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_template_translation(self):
+        self.env['res.lang']._activate_lang('fr_FR')
+        self.env.ref('base.module_base')._update_translations(['fr_FR'])
+
+        partner = self.env['res.partner'].create({'name': "test", 'lang': 'fr_FR'})
+        email_template = self.env['mail.template'].create({
+            'name': 'TestTemplate',
+            'subject': 'English Subject',
+            'body_html': '<p>English Body</p>',
+            'model_id': self.env['ir.model']._get(partner._name).id,
+            'lang': '${object.lang}'
+        })
+        self.env['ir.translation'].create({
+            'type': 'model',
+            'name': 'mail.template,subject',
+            'module': 'mail',
+            'lang': 'fr_FR',
+            'res_id': email_template.id,
+            'value': 'French Subject',
+            'state': 'translated',
+        })
+        self.env['ir.translation'].create({
+            'type': 'model',
+            'name': 'mail.template,body_html',
+            'module': 'mail',
+            'lang': 'fr_FR',
+            'res_id': email_template.id,
+            'value': '<p>French Body</p>',
+            'state': 'translated',
+        })
+        view = self.env['ir.ui.view'].create({
+            'name': 'test_layout',
+            'key': 'test_layout',
+            'type': 'qweb',
+            'arch_db': '<body><t t-raw="message.body"/> English Layout</body>'
```

Comment:
Actually does not seem to test that model_description is effectively translated. Could be included in template to test a complete translation scenario :)